### PR TITLE
Set perms on control files per Ubuntu Software Center's lintian checks.

### PR DESCRIPTION
**Pull request by yzl**
Fixes https://github.com/chef/chef-dk/issues/457.

cc @chef/omnibus-maintainers

### CODE REVIEW

File: spec/support/matchers.rb [hunk 1]
```diff
@@ -21,6 +21,14 @@
   end
 end
 
+# expect('/path/to/file').to have_permissions
+RSpec::Matchers.define :have_permissions do |perm|
```
Comment (sethvargo) [inline spec/support/matchers.rb]: This is super nice! I like! :+1:

File: spec/unit/packagers/deb_spec.rb [hunk 2]
```diff
@@ -167,13 +167,18 @@ module Omnibus
...
-      it 'copies the scripts into the DEBIAN dir' do
+      it 'copies the scripts into the DEBIAN dir with permissions = 100755' do
         subject.write_scripts
 
         expect("#{staging_dir}/DEBIAN/preinst").to be_a_file
         expect("#{staging_dir}/DEBIAN/postinst").to be_a_file
         expect("#{staging_dir}/DEBIAN/prerm").to be_a_file
         expect("#{staging_dir}/DEBIAN/postrm").to be_a_file
+
+        expect("#{staging_dir}/DEBIAN/preinst").to have_permissions '100755'
```
Comment (sethvargo) [inline spec/unit/packagers/deb_spec.rb]: Isn't the 10 bit controlled by the umask? It's possible for this to fail on other operating systems that have a different default umask.

**Comment by sethvargo** [comment]
Very elegant solution! Just one question about umask

**Comment by schisamo** [comment]
:+1:

File: spec/unit/packagers/deb_spec.rb (hunk 2 above)
Comment (danielsdeleo) [inline spec/unit/packagers/deb_spec.rb]: I think that's a regular file. `man 2 stat` on my mac has this about 1/2 way through:

```
     #define        S_IFREG  0100000  /* regular */
```

**Comment by danielsdeleo** [comment]
:+1:

File: spec/unit/packagers/deb_spec.rb (hunk 2 above)
Comment (sethvargo) [inline spec/unit/packagers/deb_spec.rb]: :+1:

**Comment by jaym** [comment]
:+1:

**Pull request merged**